### PR TITLE
Add a system property to forcibly format everything

### PR DESCRIPTION
**Pull request by pugnascotia**
In order to make it easier to migrate to an automatically-formatted world, add a system property that causes Spotless to format every Java project, ignoring the exclusion list.

### CODE REVIEW

File: build-tools-internal/src/main/groovy/elasticsearch.formatting.gradle [hunk 1]
```diff
@@ -196,7 +196,7 @@ def projectPathsToExclude = [
 
 subprojects {
   plugins.withType(ElasticsearchJavaPlugin).whenPluginAdded {
-    if (projectPathsToExclude.contains(project.path) == false) {
+    if (projectPathsToExclude.contains(project.path) == false || System.getProperty("es.format.everything") != null) {
```
Comment (breskeby) [inline build-tools-internal/src/main/groovy/elasticsearch.formatting.gradle:199]: we should use `providers.systemProperty("es.format.everything").forUseAtConfigurationTime().isPresent()`
and not access system properties directly in gradle scripts

**Review by breskeby — CHANGES_REQUESTED** [review]
We should resolve system properties via gradle provider api. 

This way gradle can track inputs of the build. Ultimately we want our build to be able to track configuration inputs to allow gradle to cache the whole configured gradle model (aka using `--configuration-cache` This will make the configuration phase way faster for repeated builds with the same inputs.

**Review by mark-vieira — APPROVED** [review]
:+1:

**Comment by pugnascotia** [comment]
Backported to `7.x` in 8579394e77e.

**Pull request merged**